### DISCUSSION TITLE
Reduce the requests for Avatar images

### DISCUSSION THREAD

**Discussion by sbaechler** [radix-ui/primitives#3213] [Ideas] [answered]
Currently, the Avatar component does a Ajax request to check if the avatar image is available and to decide if it should show the image or the fallback.

Right now there is no caching of any kind. So when you do a navigation, the request is sent again. This is especially problematic if you have a list of avatars.

I think it would be helpful if we could check if the image was in the browser cache before doing the request.

**Comment by lauri865** [comment]
Add cache-control headers, and your avatars will be served from memory/disk cache on subsequent requests. Alternatively, use a service-worker to handle it on client-side.

**Reply by sbaechler** [reply]
Radix-UI uses a `cache-control: no-cache` header which forces the browser to always revalidate the image, even if the image server responds with long cache headers.

Here is an example from the [shadcn/ui website](https://ui.shadcn.com/docs/components/avatar).
<img width="677" alt="Bildschirmfoto 2024-11-21 um 13 53 42" src="https://github.com/user-attachments/assets/2e31f627-7a83-4b00-80d2-ae9ae523e837">

I guess the issue could be solved if the `Cache-Control` request header was removed.

**Reply by lauri865** [reply, marked as answer]
I guess you have "disable cache" turned on in the devtools. Radix doesn't set any headers.

**Reply by sbaechler** [reply]
I just noticed. 🤦‍♂️ When I remove the checkbox, everything is fine.

<img width="460" alt="Bildschirmfoto 2024-11-21 um 14 10 14" src="https://github.com/user-attachments/assets/d218d9e3-7ba9-4f77-9767-92bb2582b004">



We can close this issue then.
